be/src/storage/lake/delta_writer.h

```diff
@@ -128,7 +102,94 @@ class DeltaWriter {
     void TEST_set_miss_auto_increment_column();
 
 private:
+    DeltaWriter(TabletManager* tablet_manager, int64_t tablet_id, int64_t txn_id, int64_t partition_id,
+                const std::vector<SlotDescriptor*>* slots, const std::string& merge_condition,
+                bool miss_auto_increment_column, int64_t table_id, int64_t immutable_tablet_size,
+                MemTracker* mem_tracker, int64_t max_buffer_size);
+
     DeltaWriterImpl* _impl;
 };
 
+class DeltaWriterBuilder {
```

Comment:
This is flexible, but the downside is, given X parameters there could be 2^X combinations to construct the object, with which, there could be y combinations are valid. Others are all invalid. This builder doesn't look like take care of this situation, but assuming all the combinations are valid.

Reply:
Yes, I totally agree with you, I just haven't come up with a better solution, do you have any suggestions?

Reply:
maybe, make the target object constructor private, and move the check into Builder.build() with exception or nullptr.
or add some convenient interface to builder such as,
* Builder.setX().setY().buildForXXX(a, b, c)
* Builder.setX().setY().buildForYYY(d, e, f)

Reply:
Good suggestion, I can add some checks in Build.build().